### PR TITLE
Fix top BC issues for EDMF 

### PR DESCRIPTION
**Pull request by yairchn**
This PR fixes the bug in the top BCs for EDMF in response to: 
https://github.com/CliMA/ClimateMachine.jl/issues/1743

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
Closes #1743

File: experiments/AtmosLES/bomex_model.jl [hunk 1]
```diff
@@ -465,7 +465,12 @@ function bomex_model(
                 moisture = moisture_bc,
                 turbconv = turbconv_bcs(turbconv)[1],
             ),
-            AtmosBC(turbconv = turbconv_bcs(turbconv)[2]),
+            AtmosBC(
+                momentum = Impenetrable(FreeSlip()),
+                energy = energy_bc,
```
Comment (charleskawczynski) [inline experiments/AtmosLES/bomex_model.jl:470]: Won't this add the same energy BC at the top as is at the bottom?

**Review by ilopezgp — APPROVED** [review]

**Comment by ilopezgp** [comment]
LGTM

**Comment by ilopezgp** [comment]
Actually, I think we should set the vertical velocities to zero in the top as well, not only the flux.

**Review by charleskawczynski — APPROVED** [review]

**Pull request merged**